### PR TITLE
Adjust to OTP21

### PR DESCRIPTION
**Pull request by elbrujohalcon**
At least, we need to support `gen_server:handle_continue/2`, but maybe more things.

### CODE REVIEW

**Comment by elbrujohalcon** [comment]
@getong / @michalwski : Do you have time for a code review, by any chance? 👀

File: test/wpool_process_SUITE.erl [hunk 1]
```diff
@@ -104,24 +106,87 @@ info(_Config) ->
...
+  %% handle_continue/2 returns {continue, ...}
+  SecondContinueResponse = C(continue_state_5),
+  FirstContinueResponse = {noreply, another_state, {continue, SecondContinueResponse}},
+  CastResponse = {noreply, state, {continue, FirstContinueResponse}},
+  wpool_process:cast(Pid, CastResponse),
+  continue_state_5 = get_state(Pid),
+
+  %% handle_continue/2 returns timeout = 0
+  wpool_process:cast(Pid, {noreply, state, {continue, {noreply, continue_state_7, 0}}}),
+  timer:sleep(100),
```
Comment (michalwski) [inline test/wpool_process_SUITE.erl]: From my experience sleeping arbitrary time may easily fail on hosted CIs like travis. What we recently try to do in esl/MongooseIM is a busy wait with retries. In our tests we have a function [wait_until/2 or /4](https://github.com/esl/MongooseIM/blob/7ea42e809b0e4eb77b8df1342e9c434f1047b67b/big_tests/tests/mongoose_helper.erl#L275) which takes a fun to execute and expected value. The fun is called and `wait_until` returns when the result matches expected value. If not, the `wait_until` function waits some time (by default 100ms) and retries until a deadline pass (by default 5s).

This is just a comment not a request for changes.

File: test/wpool_process_SUITE.erl [hunk 2]
```diff
@@ -79,7 +81,7 @@ init_timeout(_Config) ->
   {ok, Pid} =
     wpool_process:start_link(?MODULE, echo_server, {ok, state, 0}, []),
   timer:sleep(1),
-  timeout = wpool_process:call(?MODULE, state, 5000),
+  timeout = get_state(?MODULE),
```
Comment (michalwski) [inline test/wpool_process_SUITE.erl]: I find it hard to understand why this change works. Could you explain it to me?

File: test/wpool_process_SUITE.erl (hunk 1 above)
Comment (michalwski) [reply test/wpool_process_SUITE.erl]: And here's the random fail on travis due to waiting to short (as far as I can tell).

```
%%% wpool_process_SUITE ==> info: FAILED
%%% wpool_process_SUITE ==> {{badmatch,newerstate},
 [{wpool_process_SUITE,info,1,
                       [{file,"/home/travis/build/inaka/worker_pool/_build/test/lib/worker_pool/test/wpool_process_SUITE.erl"},
                        {line,96}]},
  {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
  {test_server,run_test_case_eval1,6,[{file,"test_server.erl"},{line,1045}]},
  {test_server,run_test_case_eval,9,[{file,"test_server.erl"},{line,977}]}]}
```

This happened to my PR #145. When I restarted the job, it worked just fine.

Comment (elbrujohalcon) [reply test/wpool_process_SUITE.erl]: Created #146 for that :)

File: test/wpool_process_SUITE.erl (hunk 2 above)
Comment (elbrujohalcon) [reply test/wpool_process_SUITE.erl]: I just added a proper test and docs on 9ce28d5.
Thanks for the review :)

**Comment by michalwski** [comment]
I think the PR is good to be merged now. Do we wait for @jfacorro review? I see you requested a review form him.

**Comment by elbrujohalcon** [comment]
Not specifically… but let's wait at least until 24h after PR creation for people in different TZs to review :)
